### PR TITLE
Address some TODOs

### PR DESCRIPTION
**Pull request by jveski**
- Add integration test for pod concurrency limits
- Add timeouts to reconciler operations to avoid deadlocks
- Configure CEL runtime to support context interruptions
- Replace the lazy/ugly/racy time.Sleep call from the exec controller with a better synchronization mechanism

### CODE REVIEW

**Comment by AYM1607** [comment]
> lazy/ugly/racy time.Sleep

So many nasty adjectives for a poor function call 😢

**Comment by jveski** [comment]
> > lazy/ugly/racy time.Sleep
> 
> So many nasty adjectives for a poor function call 😢

lol I'm so sick of race conditions 🤮

File: cmd/eno-reconciler/main.go [hunk 1]
```diff
@@ -108,7 +110,7 @@ func run() error {
 	writeBuffer := flowcontrol.NewResourceSliceWriteBufferForManager(mgr, writeBatchInterval, 1)
 
 	rCache := reconstitution.NewCache(mgr.GetClient())
-	reconciler, err := reconciliation.New(mgr, rCache, writeBuffer, remoteConfig, discoveryMaxRPS, rediscoverWhenNotFound, readinessPollInterval)
+	reconciler, err := reconciliation.New(mgr, rCache, writeBuffer, remoteConfig, discoveryMaxRPS, rediscoverWhenNotFound, timeout, readinessPollInterval)
```
Comment (AYM1607) [inline cmd/eno-reconciler/main.go:113]: Time to put these options in a reconciler specific struct or not yet? 👀

Comment (jveski) [reply cmd/eno-reconciler/main.go:113]: Personally I think it's right on the threshold, like the next one definitely needs it.

File: internal/controllers/synthesis/exec.go [hunk 2]
```diff
@@ -91,13 +104,54 @@ func (c *execController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, fmt.Errorf("updating composition status: %w", err)
 	}
 
-	// Let the informers catch up
-	// Obviously this isn't ideal, consider a lamport clock in memory
-	time.Sleep(time.Millisecond * 100)
+	// Reset the debounce annotation after synthesis, since the process likely took
+	// longer than the initial debounce period
+	pod.Annotations[execStartAnnotation] = time.Now().Format(time.RFC3339)
```
Comment (AYM1607) [inline internal/controllers/synthesis/exec.go:109]: `time.Now().UTC()`?

File: cmd/eno-reconciler/main.go (hunk 1 above)
Comment (jveski) [reply cmd/eno-reconciler/main.go]: Thought about it more and pushed a commit to add one 🙃

**Review by AYM1607 — APPROVED** [review]
LGTM

**Pull request merged**